### PR TITLE
Feature/40014

### PR DESCRIPTION
**Pull request by anmolgarggit**

### CODE REVIEW

File: src/config/configuration.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ const configuration: IConfig = Object.freeze({
     password: envVars.PASSWORD,
     port: envVars.PORT,
     secretKey: envVars.SECRET_KEY,
+    role:envVars.role
```
Comment (mohit-nx) [inline src/config/configuration.ts]: linting

File: src/controllers/trainee/Conroller.ts [hunk 2]
```diff
@@ -1,56 +1,117 @@
...
+            if (email != null && email != "") {
+                data['email'] = email;
+            }
+            if (password != null && password != "") {
+                const saltRounds = 15;
+                const salt = bcrypt.genSaltSync(saltRounds);
+                const hash = bcrypt.hashSync(password, salt);
+                data['password'] = hash;
+            }
+            if (name != null && name != "") {
```
Comment (mohit-nx) [inline src/controllers/trainee/Conroller.ts]: We do not require check name != "". It should be handled by the validations.

File: src/controllers/trainee/routes.ts [hunk 3]
```diff
@@ -6,13 +6,13 @@ import validation from './validation';
...
 
-traineeRoute.route('/:id?')
-    .get(validationHandler(validation.get), traineeController.get)
-    // .get(validationHandler(validation.get),authMiddleware('getUsers','read'),traineeController.get)
-    .post(validationHandler(validation.create), traineeController.post)
-    .put(validationHandler(validation.update), traineeController.put)
-    .delete(validationHandler(validation.delete), traineeController.delete);
+traineeRoute.route('/')
+    //.get( traineeController.get)
+    traineeRoute.route('/all').get(validationHandler(validation.get),authMiddleware('getUsers','all'),traineeController.get)
```
Comment (mohit-nx) [inline src/controllers/trainee/routes.ts]: We can cascade the routes.

File: src/controllers/trainee/routes.ts [hunk 4]
```diff
@@ -6,13 +6,13 @@ import validation from './validation';
...
+    traineeRoute.route('/create').post(validationHandler(validation.create),authMiddleware('getUsers','all'), traineeController.post)
+    traineeRoute.route('/update').put(validationHandler(validation.update),authMiddleware('getUsers','all'), traineeController.put)
```
Comment (mohit-nx) [inline src/controllers/trainee/routes.ts]: the permission should be different for different routes

**Comment by nxvinay** [comment]
@anmolgarggit @mohit-nx What is the status of this PR?

**Pull request merged**